src/clients/AcrossHubPoolClient.ts

```diff
@@ -0,0 +1,629 @@
+    );
+    this.executedRootBundles.push(
+      ...executedRootBundleEvents
+        .filter((event) => !this.config.ignoredHubExecutedBundles.includes(event.blockNumber))
+        .map((event) => spreadEventWithBlockNumber(event) as ExecutedRootBundle)
+    );
+
+    // If the contract's current rootBundleProposal() value has an unclaimedPoolRebalanceLeafCount > 0, then
+    // it means that either the root bundle proposal is in the challenge period and can be disputed, or it has
+    // passed the challenge period and pool rebalance leaves can be executed. Once all leaves are executed, the
+    // unclaimed count will drop to 0 and at that point there is nothing more that we can do with this root bundle
+    // besides proposing another one.
+    if (pendingRootBundleProposal.unclaimedPoolRebalanceLeafCount > 0) {
+      const mostRecentProposedRootBundle = this.proposedRootBundles[this.proposedRootBundles.length - 1];
+      this.pendingRootBundle = {
+        poolRebalanceRoot: pendingRootBundleProposal.poolRebalanceRoot,
+        relayerRefundRoot: pendingRootBundleProposal.relayerRefundRoot,
+        slowRelayRoot: pendingRootBundleProposal.slowRelayRoot,
+        proposer: pendingRootBundleProposal.proposer,
+        unclaimedPoolRebalanceLeafCount: pendingRootBundleProposal.unclaimedPoolRebalanceLeafCount,
+        challengePeriodEndTimestamp: pendingRootBundleProposal.challengePeriodEndTimestamp,
+        bundleEvaluationBlockNumbers: mostRecentProposedRootBundle.bundleEvaluationBlockNumbers.map(
+          (block: BigNumber) => {
+            // Ideally, the HubPool.sol contract should limit the size of the elements within the
+            // bundleEvaluationBlockNumbers array. But because it doesn't, we wrap the cast of BN --> Number
+            // in a try/catch statement and return some value that would always be disputable.
+            // This catches the denial of service attack vector where a malicious proposer proposes with bundle block
+            // evaluation block numbers larger than what BigNumber::toNumber() can handle.
+            try {
+              return block.toNumber();
+            } catch {
+              return 0;
+            }
+          }
+        ),
+        proposalBlockNumber: mostRecentProposedRootBundle.blockNumber,
+      };
+    } else {
+      this.pendingRootBundle = undefined;
+    }
+
+    this.isUpdated = true;
+    this.firstBlockToSearch = searchConfig.toBlock + 1; // Next iteration should start off from where this one ended.
+
+    this.logger.debug({ at: "HubPoolClient", message: "HubPool client updated!" });
+  }
+
+  private async fetchTokenInfoFromContract(address: string): Promise<L1Token> {
+    const token = new Contract(address, ERC20__factory.abi, this.hubPool.signer);
+    const [symbol, decimals] = await Promise.all([token.symbol(), token.decimals()]);
+    return { address, symbol, decimals };
+  }
```

Comment:
IMO it'd be nice if we could factor generic methods like this one out of the HubPoolClient before migration. In this case it _probably_ belongs in the SDK, but for any instances where it belongs in relayer-v2, we'd have to migrate it back over the SDK-v2/relayer-v2 boundary.